src/style-spec/expression/index.js

```diff
@@ -107,8 +141,9 @@ function createExpression(expression: mixed, expectedType: Type | null): StyleEx
     };
 }
 
-function createExpressionWithErrorHandling(expression: mixed, expectedType: Type | null, defaultValue: Value | null): StyleExpression {
-    expression = createExpression(expression, expectedType);
+function createExpressionWithErrorHandling(expression: mixed, options: StyleExpressionOptions): StyleExpression {
+    expression = createExpression(expression, options);
+    const defaultValue = options.defaultValue === undefined ? null : options.defaultValue;
```

Comment:
Should we have `errorHandling: boolean` as an option and then expose only `createExpression()`?

Reply:
Yeah, we need to unify the two somehow for #5222 and #5370. I was trying to figure out a clean way to do it. Could we use an error return type by default, and push the try/catch up to `StyleDeclaration#calculate`? I avoided doing this in previous performance PRs because it would make the expression benchmarks not strictly comparable, but I'm 👍  on doing it in an independent PR.

Also, we should use `Color` ubiquitously so `unwrap` becomes unnecessary.

Reply:
> Could we use an error return type by default, and push the try/catch up to `StyleDeclaration#calculate`?

Hmm, nope, that regresses layout performance, presumably because constant "evaluation" is also wrapped. https://bl.ocks.org/anonymous/raw/2b247bab7e59c3f83fa75ad8c7f2921d/

Reply:
@jfirebaugh could we just decide whether to do error handling based on the presence or absence of `defaultValue`